ui/components/grants-list.js

```diff
@@ -0,0 +1,41 @@
+import { sortByResource } from '../lib/grants'
+
+import RoleSelect from './role-select'
+
+export default function GrantsList({ grants, mutate }) {
```

Comment:
Would it be possible to move the grant fetching and mutating within this component? The props would then be:

```
({ destination })
```

Reply:
hmm we definitely can. However, we are also using the `grants` in the parent component to determine if the grant list is empty or not - which it has a specific margin ( which we don't want to include that in the component) and also for the user sidebar - we also have inherited list as well. We can pass the inherited list as a children prop but I am not very confident that is a good idea.

Reply:
We can keep it as: passing `mutate` is definitely something we should try to improve though since it isn't obvious what that is – I would suggest changing this to `onChange` and `onRemove` and handling edits in the parent component since that's where the data is coming from